SampleApp/Program.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Text.Json;
+
+namespace SampleApp
+{
+    public class Program
+    {
+        /// <summary>
+        /// Session key for the cached forecast data
+        /// </summary>
+        private const string ForecastCacheKey = "forecast";
+
+        public static void Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+
+            // Add services to the container.
+            builder.Services.AddAuthorization();
+
+            // Add session state support
+            builder.Services.AddSession(options =>
+            {
+                // The following option will cache weather forecasts for 30 seconds.
+                // Reloading a cached forecast will reset this timer. If a cached forecast
+                // is not requested for 30 seconds it should expire and a new
+                // forecast will be generated.
+                options.IdleTimeout = TimeSpan.FromSeconds(30);
+                options.Cookie.IsEssential = true;
```

Comment:
just a question here, but why is the cookie essential here?

Reply:
https://andrewlock.net/session-state-gdpr-and-non-essential-cookies/ is a nice overview, but essentially the cookie is tied to the session ID which is used as the key for the data in the distributed cache.

`IsEssential` may be overkill because it's recommended that applications should work without cached data, and it may be at odds with GDPR, but it ensures the sample works.